Source/WebCore/page/scrolling/mac/ScrollerPairMac.mm

```diff
@@ -145,27 +154,30 @@ - (void)scrollerImpPair:(NSScrollerImpPair *)scrollerImpPair updateScrollerStyle
 
     [m_scrollerImpPairDelegate invalidate];
     [m_scrollerImpPair setDelegate:nil];
+
+    ensureOnMainThread([scrollerImpPair = std::exchange(m_scrollerImpPair, nil)] {
+    });
 }
 
 void ScrollerPairMac::handleWheelEventPhase(PlatformWheelEventPhase phase)
 {
-    Locker locker { m_scrollerImpPairLock };
-
-    switch (phase) {
-    case WebCore::PlatformWheelEventPhase::Began:
-        [m_scrollerImpPair beginScrollGesture];
-        break;
-    case WebCore::PlatformWheelEventPhase::Ended:
-    case WebCore::PlatformWheelEventPhase::Cancelled:
-        [m_scrollerImpPair endScrollGesture];
-        break;
-    case WebCore::PlatformWheelEventPhase::MayBegin:
-        [m_scrollerImpPair beginScrollGesture];
-        [m_scrollerImpPair contentAreaScrolled];
-        break;
-    default:
-        break;
-    }
+    ensureOnMainThreadWithImpPair([phase](NSScrollerImpPair *impPair) {
```

Comment:
Do we need to retain the impPair? Or do we need to pass it at all, since `ensureOnMainThreadWithImpPair` has access to |this| on the main thread?

Reply:
Ah, so passing it in here was just a convenience (though it's true that `this` is safe to access, so I can just use `m_scrollerImpPairLock` directly) — I will remove the `NSScrollerImpPair *` argument and just use `m_scrollerImpPairLock`.